recipes/libxpm/all/conanfile.py

```diff
@@ -0,0 +1,71 @@
+import functools
+import os
+
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.43.0"
+
+
+class LibXpmConan(ConanFile):
+    name = "libxpm"
+    description = "X Pixmap (XPM) image file format library"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://gitlab.freedesktop.org/xorg/lib/libxpm"
+    topics = "xpm"
+    license = "MIT-open-group"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+    generators = "cmake"
+    exports_sources = "CMakeLists.txt", "windows/*"
+    no_copy_source = True
+
+    def validate(self):
+        if self.settings.os not in ("Windows", "Linux", "FreeBSD"):
+            raise ConanInvalidConfiguration(
+                "libXpm is supported only on Windows, Linux and FreeBSD"
+            )
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True)
```

Comment:
Typically we prefer out of source builds

https://github.com/conan-io/conan-center-index/blob/9f21b791bce36c330be3fc06dd4c6a8aaa5cf3a5/recipes/mfast/all/conanfile.py#L91-L93